File: app/src/main/java/me/tagavari/airmessage/fragment/FragmentCommunication.java
package me.tagavari.airmessage.fragment;

import androidx.fragment.app.Fragment;

public class FragmentCommunication<A> extends Fragment {
	private A callback;
	
	public void setCommunicationsCallback(A callback) {
		this.callback = callback;
	}
	
	protected A getCommunicationsCallback() {
		return callback;
	}
}

File: app/src/main/java/me/tagavari/airmessage/fragment/FragmentCommunication.kt
package me.tagavari.airmessage.fragment

import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment

open class FragmentCommunication<A: Any> : Fragment {
	var communicationsCallback: A? = null
	
	constructor() : super()
	constructor(@LayoutRes contentLayoutId: Int) : super(contentLayoutId)
}